pkg/prototype/values_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 The ksonnet authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package prototype
+
+import (
+	"testing"
+
+	ksstrings "github.com/ksonnet/ksonnet/pkg/util/strings"
+	"github.com/ksonnet/ksonnet/pkg/util/test"
+	"github.com/spf13/afero"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValues(t *testing.T) {
+	proto := &Prototype{
+		Params: ParamSchemas{
+			{
+				Name: "name",
+				Type: String,
+			},
+			{
+				Name: "key1",
+				Type: String,
+			},
+			{
+				Name:    "key2",
+				Type:    String,
+				Default: ksstrings.Ptr("default"),
+			},
+		},
+	}
+
+	cases := []struct {
```

Comment:
We could use one more test to show precedence logic between bare flags and values file.